requestor-tutorials/service-development/service-example-2-managed-erigon.md

```diff
@@ -40,5 +40,361 @@ A typical interaction flow proceeds in the following manner:
+
+{% hint style="info" %}
+We have no control of the `Erigon` service class initialization, so the
+
+```python
+def __init__(self, *args, **kwargs):
+    super().__init__(*args, **kwargs)
+```
+
+part is required.
+{% endhint %}
+
+#### Payload
+
+The payload is defined in [ErigonPayload](https://github.com/golemfactory/yagna-service-erigon/blob/master/requestor/server/erigon_payload.py) class:
+
+```python
+ERIGON_RUNTIME_NAME = "erigon"
+
+@dataclass
+class ErigonPayload(Payload):
+    runtime: str = constraint(inf.INF_RUNTIME_NAME, "=", ERIGON_RUNTIME_NAME)
+    min_mem_gib: float = constraint(inf.INF_MEM, ">=", 0.5)
+```
+
+and used in `get_payload` in the service:
+
+```python
+@classmethod
+async def get_payload(cls):
+    return ErigonPayload()
+```
+
+Few important things to note here:
+
+* We don't have any `image_hash` (contrary to the previouse examples) because we don't use a VM-based runtime
+* We declare the runtime name, `erigon` - this must match the offered [exeunit-name](#pluging-the-runtime-into-golemsp)
+* `min_mem_gib` - minimum amount of RAM the provider has to offer. This is pretty useless in the Erigon case (contrary to the e.g. VM-based runtimes), but is required because of a known `yapapi` [bug](https://github.com/golemfactory/yapapi/issues/500).
+
+
+#### Start
+
+```python
+async def start(self):
+    #   startup - set start args (network parameter)
+    self._ctx.deploy()
+    start_args = await self._get_start_args()
+    if start_args:
+        erigon_init_args = start_args[0]
+        erigon_init_args_str = json.dumps(erigon_init_args)
+        self._ctx.start(erigon_init_args_str)
+    else:
+        self._ctx.start()
+
+    #   Set url & auth
+    self._ctx.run('STATUS')
+    processing_future = yield self._ctx.commit()
+    result = self._parse_status_result(processing_future.result())
+    self.url, self.auth, self.network = result['url'], result['auth'], result['network']
+```
+
+Let's split that to separate parts.
+
+##### Deploy
+
+```python
+self._ctx.deploy()
+```
+
+This is the first thing that should always be done with `self._ctx`.
+
+##### Determine start args
+
+```python
+start_args = await self._get_start_args()
+```
+
+`start_args` - in the erigon case - are a single-item tuple:
+
+```python
+(
+    {'network': <ETHEREUM NETWORK NAME>},
+)
+```
+
+This `start_args` are defined by the final user (the one ordering Erigon service, e.g. via the web interface) and passed directly to the runtime.
+
+{% hint style="info" %}
+Current `yapapi` has no pretty way of passing arguments to the `Service`, so this is implemented as a ugly-but-harmless hack:
+
+```python
+async def _get_start_args(self):
+    while True:
+        try:
+            return self._cluster.instance_start_args
+        except AttributeError:
+            await asyncio.sleep(0.1)
+```
+
+This will be fixed in [the near future](https://github.com/golemfactory/yapapi/issues/372).
```

Comment:
```suggestion
This will be addressed in [the near future](https://github.com/golemfactory/yapapi/issues/372).
```

I'd say "addressed", not "fixed" since it's not a "bug" per se...